hphp/hack/src/parser/parser_trait.rs

```diff
@@ -736,6 +741,10 @@ where
         self.lexer_mut().next_xhp_class_name()
     }
 
+    fn next_xhp_class_name_not_qualified(&mut self) -> S::Token {
```

Comment:
I find the term 'not qualified' a little confusing here. How about something like `next_xhp_modifier_class_name` to make it clear you're accepting the syntax `xhp class Foo...`?

A comment (perhaps on `next_xhp_class_name_no_qualifier` in lexer.rs) with an example of the expected syntax would be super helpful too.